executor/aggfuncs/func_var_pop.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pkg/errors"
+)
+
+// All the following avg function implementations return the float64 result,
+// which store the partial results in "partialResult4VarPopFloat64".
+//
+// "baseVarPopFloat64" is wrapped by:
+// - "varPopOriginal4Float64"
+// - "varPopPartial4Float64"
+type baseVarPopFloat64 struct {
+	baseAggFunc
+}
+
+type partialResult4VarPopFloat64 struct {
+	sum       float64
+	squareSum float64
+	count     int64
+}
+
+func (e *baseVarPopFloat64) AllocPartialResult() PartialResult {
+	return (PartialResult)(&partialResult4VarPopFloat64{})
+}
+
+func (e *baseVarPopFloat64) ResetPartialResult(pr PartialResult) {
+	p := (*partialResult4VarPopFloat64)(pr)
+	p.sum, p.squareSum, p.count = 0, 0, 0
+}
+
+func (e *baseVarPopFloat64) AppendFinalResult2Chunk(sctx sessionctx.Context, pr PartialResult, chk *chunk.Chunk) error {
+	p := (*partialResult4VarPopFloat64)(pr)
+	if p.count == 0 {
+		chk.AppendNull(e.ordinal)
+	} else {
+		chk.AppendFloat64(e.ordinal, (p.squareSum-p.sum*p.sum/float64(p.count))/float64(p.count))
```

Comment:
Please add a comment for this formula.

Reply:
see http://types/helper.go#CalculateMerge